opal/corelib/runtime.js

```diff
@@ -2445,7 +2445,7 @@
     if (typeof str === 'string')
       throw Opal.FrozenError.$new("can't modify frozen String");
 
-    var encoding = Opal.encodings[name];
+    var encoding = Opal.Encoding.$find(name);
```

Comment:
Another way to fix this issue is to call `toUpperCase()` on `name`.
For reference, `Opal.encodings` contains:

```js
{
  'UTF-8': constructor {
    '$$id': 14,
    name: 'UTF-8',
    names: [ 'UTF-8', 'CP65001' ],
    ascii: true,
    dummy: false
  },
  CP65001: constructor {
    '$$id': 14,
    name: 'UTF-8',
    names: [ 'UTF-8', 'CP65001' ],
    ascii: true,
    dummy: false
  },
  'UTF-16LE': constructor {
    '$$id': 22,
    name: 'UTF-16LE',
    names: [ 'UTF-16LE' ],
    ascii: false,
    dummy: false
  },
  'UTF-16BE': constructor {
    '$$id': 28,
    name: 'UTF-16BE',
    names: [ 'UTF-16BE' ],
    ascii: false,
    dummy: false
  },
  'UTF-32LE': constructor {
    '$$id': 34,
    name: 'UTF-32LE',
    names: [ 'UTF-32LE' ],
    ascii: false,
    dummy: false
  },
  'ASCII-8BIT': constructor {
    '$$id': 40,
    name: 'ASCII-8BIT',
    names: [ 'ASCII-8BIT', 'BINARY', 'US-ASCII', 'ASCII' ],
    ascii: true,
    dummy: true
  },
  BINARY: constructor {
    '$$id': 40,
    name: 'ASCII-8BIT',
    names: [ 'ASCII-8BIT', 'BINARY', 'US-ASCII', 'ASCII' ],
    ascii: true,
    dummy: true
  },
  'US-ASCII': constructor {
    '$$id': 40,
    name: 'ASCII-8BIT',
    names: [ 'ASCII-8BIT', 'BINARY', 'US-ASCII', 'ASCII' ],
    ascii: true,
    dummy: true
  },
  ASCII: constructor {
    '$$id': 40,
    name: 'ASCII-8BIT',
    names: [ 'ASCII-8BIT', 'BINARY', 'US-ASCII', 'ASCII' ],
    ascii: true,
    dummy: true
  }
}
```

Reply:
That's a good point :+1: